test/functional/connection-tests.py

```diff
@@ -162,16 +182,121 @@ def test_bind_no_args(self, mock_conn, mock_drift, mock_validate):
+        self.assertTrue(result is not None)
+        self.assertTrue('id' in result)
+        self.assertTrue('statusPath' in result)
+
+        job_result = self._wait_for_job(result['id'])
+        self._verify_job(job_result)
+
+        # Note: no data is returned from the request,
+        # need to verify that some other way
+        compliance = self.cp.getCompliance(self.consumer_uuid)
+        self.assertTrue('compliantProducts' in compliance)
+        compliant_products = compliance['compliantProducts']
+        self.assertTrue(test_product in compliant_products)
+        self.assertEqual(compliance['status'], 'valid')
+
+    def test_async_bind_by_pool(self):
+        pools = self.cp.getPoolsList(self.consumer_uuid)
+        installed_product = '37060'
+        product_to_pool = {}
+        for pool in pools:
+            provided_products = pool['providedProducts']
+            for provided_product in provided_products:
+                # just keep the last one
+                product_to_pool[provided_product['productId']] = pool['id']
+
+        target_pool = product_to_pool[installed_product]
+        result = self.cp.bindByEntitlementPoolAsync(self.consumer_uuid,
+                                                    target_pool, '1')
+        self.assertTrue('id' in result)
+
+        job_result = self._wait_for_job(result['id'])
+        self._verify_job(job_result)
+
+        compliance = self.cp.getCompliance(self.consumer_uuid)
+        self.assertTrue('status' in compliance)
+        self.assertEqual(compliance['status'], 'valid')
+        self.assertTrue(installed_product in compliance['compliantProducts'])
+
+    def test_async_auto_bind(self):
+        result = self.cp.bindAsync(self.consumer_uuid)
+        self.assertTrue(result is not None)
+        self.assertTrue('id' in result)
+
+        job_result = self._wait_for_job(result['id'])
+
+        self._verify_job(job_result)
+
+        compliance = self.cp.getCompliance(self.consumer_uuid)
+        self.assertTrue('status' in compliance)
+        self.assertEqual(compliance['status'], 'valid')
+
+    def test_update_job_status(self):
+        result = self.cp.bindAsync(self.consumer_uuid)
+
+        updated_status = self.cp.updateJobStatus(result)
+        self.assertTrue(updated_status is not None)
+        self.assertTrue('id' in result)
+        self.assertEqual(result['id'], updated_status['id'])
+        self.assertEqual(result['targetId'], self.consumer_uuid)
+        self.assertEqual(updated_status['targetId'], self.consumer_uuid)
+        self.assertNotEqual(result['updated'], updated_status['updated'])
+
+    def _verify_job(self, job_results):
+        self.assertTrue(job_results['done'])
+        self.assertFalse(job_results['state'] == "FAILED")
+
+    # TODO: replace this with an event loop and event source
+    def _wait_for_job(self, job_id, wait_interval=2, max_attempts=6):
+        # if it's not ready in 30s, reason enough to fail the test
```

Comment:
This comment confuses me a bit.  Isn't the worst-case `wait_interval` * `max_attempts`?  I think the implementation is fine, but it'd be nice to correct/remove the comment.